litestar/middleware/session/client_side.py

```diff
@@ -117,10 +117,18 @@ def _create_session_cookies(self, data: list[bytes], cookie_params: dict[str, An
                     include={f for f in Cookie.__dict__ if f not in ("key", "secret")},
                 )
             )
+
+        def _cookie_chunk_key(i: int) -> str:
+            # If the data is split into multiple cookies, the key will be of the format `session-{segment number}`,
+            # however if only one cookie is needed, the key will be `session`.
+            if len(data) == 1:
+                return self.config.key
+            return f"{self.config.key}-{i}"
```

Comment:
I'm not sure if creating a function here is a good idea. This would result in a new function object being created everytime this method is called.

I think using an if condition directly would be enough.

```python
if len(data) == 1:
    return [Cookie(...)] # correct naming

# the existing list comprehension
...
```